docs/writing-stories/args.md

```diff
@@ -122,6 +122,37 @@ The `args` param is always a set of `key:value` pairs delimited with a semicolon
 
 Args specified through the URL will extend and override any default values of args specified on the story.
 
+## Mapping to complex arg values
+
+Complex values such as JSX elements cannot be serialized to the manager (e.g. the Controls addon) or synced with the URL. To work around this limitation, arg values can be "mapped" from a simple string to a complex type using the `mapping` property in `argTypes`. This works on any type of arg, but makes most sense when used with the 'select' control.
+
+```
+argTypes: {
+  label: {
+    control: {
+      type: 'select',
+      options: ['Normal', 'Bold', 'Italic']
+    },
+    mapping: {
+      Bold: <b>Bold</b>,
+      Italic: <i>Italic</i>
+    }
+  }
+}
+```
+
+Note that `mapping` does not have to be exhaustive. If the arg value is not a property of `mapping`, the value will be used directly. Keys in `mapping` always correspond to arg *values*, even when `options` is an object. Specifying `options` as an object (key-value pairs) is useful if you want to use special characters in the input label. For example:
```

Comment:
This is different than what we discussed. Can you explain your rationale behind the change?

Reply:
@shilman it seems kind of like what I was picturing? Which part is different?

Reply:
@shilman and I discussed setting the arg to `undefined` if no matching key is found on `mapping`. I deviated from that because it doesn't make sense when using mapping with a control that doesn't have `options`. I did implement it with different behavior for controls with `options`, but it was so confusing I decided to just avoid it altogether. Invalid `options` are already ignored anyway. I also see benefit in non-exhaustive mapping, because it allows one to map only the special variants rather than everything.

Reply:
Got it. So in the case of the a choice that doesn't have options, you can still have a mapping but it would just cover a few special cases and fall back to the default value?

Reply:
Exactly. So you can have a `text` control and a mapping that only applies for a particular value (e.g. the default value).